### PR TITLE
[FLINK-31952][table] Support 'EXPLAIN' statement for CompiledPlan.

### PR DESCRIPTION
**Pull request by xishuaidelin**
## What is the purpose of the change

* This pull request support 'EXPLAIN' statement for CompiledPlan which contributes to the FLINK-31791

## Brief change log

  - *Modify parserImpl.ftl template to add syntax support, which can be verified by FlinkSqlParserImplTest*
  - *Translate the sql to Operation*
  - *Got the internal plan from the Operation  and pass it to TableEnv#explainPlan*

## Verifying this change

This change could be verified by the testRichExplainFilePath in SqlDmlToOperationConverterTest  and testExplainCompiledPlan in TableEnvironmentTest.

## Does this pull request potentially affect one of the following parts:

  - Dependencies (does it add or upgrade a dependency): (yes / **no**)
  - The public API, i.e., is any changed class annotated with `@Public(Evolving)`: (yes / **no**)
  - The serializers: (yes / **no** / don't know)
  - The runtime per-record code paths (performance sensitive): (yes / **no** / don't know)
  - Anything that affects deployment or recovery: JobManager (and its components), Checkpointing, Kubernetes/Yarn, ZooKeeper: (yes / **no** / don't know)
  - The S3 file system connector: (yes / **no** / don't know)

## Documentation

  - Does this pull request introduce a new feature? (yes / **no**)
  - If yes, how is the feature documented? (**not applicable** / docs / JavaDocs / not documented)

### CODE REVIEW

File: flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/TableEnvironmentImpl.java [hunk 1]
```diff
@@ -915,6 +916,16 @@ public TableResultInternal executeInternal(Operation operation) {
             return executeInternal(((StatementSetOperation) operation).getOperations());
         } else if (operation instanceof ExplainOperation) {
             ExplainOperation explainOperation = (ExplainOperation) operation;
+            if (explainOperation instanceof ExplainFileOperation) {
+                ExplainFileOperation explainfileOperation = (ExplainFileOperation) explainOperation;
+                CompiledPlan cp =
+                        loadPlan(PlanReference.fromFile(explainfileOperation.getFilePath()));
```
Comment (LadyForest) [inline flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/TableEnvironmentImpl.java:922]: Why just load the plan and return it?  I think we should call `compiledPlan#explain(...)`

**Review by LadyForest — CHANGES_REQUESTED** [review]
Hi @xishuaidelin, thanks for your contribution.

I think you might misunderstand the purpose of this subtask. We should return the explanation result of the compiled plan, rather than the plan itself.

File: flink-table/flink-sql-parser/src/main/java/org/apache/flink/sql/parser/dql/SqlRichExplain.java [hunk 2]
```diff
@@ -58,6 +60,10 @@ public SqlNode getStatement() {
         return statement;
     }
 
+    public String getPlanFile() {
```
[the remark this answers is not in the document]
Comment (xishuaidelin) [reply flink-table/flink-sql-parser/src/main/java/org/apache/flink/sql/parser/dql/SqlRichExplain.java:63]: I removed this method and added this logic to `convertExplainPlanOperation`

File: flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/TableEnvironmentImpl.java (hunk 1 above)
Comment (xishuaidelin) [reply flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/TableEnvironmentImpl.java:922]: I got it and fix it with `compiledPlan#explain`

Comment (xishuaidelin) [reply flink-table/flink-table-api-java/src/main/java/org/apache/flink/table/api/internal/TableEnvironmentImpl.java:922]: Ok. I fixed it.

**Pull request closed without merging**